elide-core/src/main/java/com/yahoo/elide/jsonapi/document/processors/SparseFieldsetProcessor.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.jsonapi.document.processors;
+
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.jsonapi.models.JsonApiDocument;
+import com.yahoo.elide.jsonapi.models.Relationship;
+import com.yahoo.elide.jsonapi.models.Resource;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Implementation for JSON API 'fields' query param.
+ * Includes only the fields requested by the arguments to fields.
+ */
+public class SparseFieldsetProcessor implements DocumentProcessor {
+    private static Map<String, Set<String>> parseSparseFields(MultivaluedMap<String, String> queryParams) {
+        Map<String, Set<String>> result = new HashMap<>();
+
+        for (String key : queryParams.keySet()) {
+            if (key.startsWith("fields[") && key.endsWith("]")) {
+                String type = key.substring(7, key.length() - 1);
+
+                LinkedHashSet<String> filters = new LinkedHashSet<>();
+                for (String filterParams : queryParams.get(key)) {
+                    Collections.addAll(filters, filterParams.split(","));
+                }
+
+                if (filters.size() > 0) {
+                    result.put(type, filters);
+                }
+            }
+        }
+
+        return result;
+    }
+
+    private void filterResource(Resource resource, Map<String, Set<String>> sparseFields) {
+        Set<String> includeFields = sparseFields.containsKey(resource.getType())
```

Comment:
I believe this concept is correct, but does it cover all cases? For instance, consider I have a book containing references to many pages. Example:

```java
public class Book {
  ...
  List<Page> pages;
  ...

    public static class Page {
        ...
        String title;
        String summary;
        int chapterNum;
        ...
    } 
}
```

Then querying a book, I should be able to extract `book.pages.chapterNum`. iirc, the `getType()` operators won't necessarily produce the correct result in this case?